pkg/services/quota/quota.go

```diff
@@ -118,40 +128,107 @@ func (qs *QuotaService) getQuotaScopes(target string) ([]models.QuotaScope, erro
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.Org},
-			models.QuotaScope{Name: "user", Target: "org_user", DefaultLimit: qs.Cfg.Quota.User.Org},
+			models.QuotaScope{
+				Name:             "global",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Global.Org,
+			},
+			models.QuotaScope{
+				Name:             "user",
+				Target:           "org_user",
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.User.Org,
+			},
 		)
 		return scopes, nil
 	case "dashboard":
 		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.Dashboard},
-			models.QuotaScope{Name: "org", Target: target, DefaultLimit: qs.Cfg.Quota.Org.Dashboard},
+			models.QuotaScope{
+				Name:             "global",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Global.Dashboard,
+			},
+			models.QuotaScope{
+				Name:             "org",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Org.Dashboard,
+			},
 		)
 		return scopes, nil
 	case "data_source":
 		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.DataSource},
-			models.QuotaScope{Name: "org", Target: target, DefaultLimit: qs.Cfg.Quota.Org.DataSource},
+			models.QuotaScope{
+				Name:             "global",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Global.DataSource,
+			},
+			models.QuotaScope{
+				Name:             "org",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Org.DataSource,
+			},
 		)
 		return scopes, nil
 	case "api_key":
 		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.ApiKey},
-			models.QuotaScope{Name: "org", Target: target, DefaultLimit: qs.Cfg.Quota.Org.ApiKey},
+			models.QuotaScope{
+				Name:             "global",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Global.ApiKey,
+			},
+			models.QuotaScope{
+				Name:             "org",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Org.ApiKey,
+			},
 		)
 		return scopes, nil
 	case "session":
 		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.Session},
+			models.QuotaScope{
+				Name:             "global",
+				Target:           target,
+				TargetConditions: qs.getQuotaScopeConditions(target),
+				DefaultLimit:     qs.Cfg.Quota.Global.Session,
+			},
```

Comment:
Think we can skip this for now since not needed. TargetConditions will be empty by default